kpi/deployment_backends/kobocat_backend.py

```diff
@@ -84,35 +87,51 @@ def bulk_assign_mapped_perms(self):
             assign_applicable_kc_permissions(self.asset, user, perms)
 
     def bulk_update_submissions(
-        self, request_data: dict, requesting_user: 'auth.User'
+        self, data: dict, requesting_user: 'auth.User'
     ) -> dict:
         """
-        Allows for bulk updating of submissions proxied through kobocat. A
+        Allows for bulk updating of submissions proxied through KoBoCAT. A
         `deprecatedID` for each submission is given the previous value of
         `instanceID` and `instanceID` receives an updated uuid. For each key
         and value within `request_data`, either a new element is created on the
         submission's XML tree, or the existing value is replaced by the updated
         value.
 
         Args:
-            request_data (dict): must contain a list of `submission_ids` and at
+            data (dict): must contain a list of `submission_ids` and at
                 least one other key:value field for updating the submissions
-            requesting_user ('auth.User')
+            requesting_user (User)
 
         Returns:
             dict: formatted dict to be passed to a Response object
         """
-        payload = self.__prepare_bulk_update_payload(request_data)
-        kwargs = {'instance_ids': payload.pop('submission_ids')}
-        params = self.validate_submission_list_params(
-            requesting_user.id, format_type=INSTANCE_FORMAT_TYPE_XML,
-            **kwargs
+
+        submission_ids = self.validate_write_access_with_partial_perms(
+            user=requesting_user,
+            perm=PERM_CHANGE_SUBMISSIONS,
+            submission_ids=data['submission_ids'],
+            query=data['query'],
         )
-        submissions = list(self.__get_submissions_in_xml(**params))
+
+        # If `submission_ids` is not empty, user has partial permissions.
+        # Otherwise, they have have full access.
+        if submission_ids:
```

Comment:
Good question, but actually no, because inside the `validate_write_access_with_partial_perms` method, an empty list of submissions would raide a `PermissionDenied` exception and this code would be never reached. 

We have 3 scenarios: 
- Like described earlier, `query` or `submission_ids` is passed to `validate_write_access_with_partial_perms` which returns `PermissionDenied` if no submissions are found matching the criteria.
- Passed empty `query` and `submission_ids` to `validate_write_access_with_partial_perms`.
   1. No submissions found by `validate_write_access_with_partial_perms` will raise a `PermissionDenied`
   2. Submissions are found, it will enter the condition and modify all submissions that user is allowed to edit/delete.